<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
 <head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <title>用一个异常来恢复 fiber 的执行</title>
<link media="all" rel="stylesheet" type="text/css" href="styles/03e73060321a0a848018724a6c83de7f-theme-base.css" />
<link media="all" rel="stylesheet" type="text/css" href="styles/03e73060321a0a848018724a6c83de7f-theme-medium.css" />

 </head>
 <body class="docs"><div class="navbar navbar-fixed-top">
  <div class="navbar-inner clearfix">
    <ul class="nav" style="width: 100%">
      <li style="float: left;"><a href="fiber.resume.html">« Fiber::resume</a></li>
      <li style="float: right;"><a href="fiber.getreturn.html">Fiber::getReturn »</a></li>
    </ul>
  </div>
</div>
<div id="breadcrumbs" class="clearfix">
  <ul class="breadcrumbs-container">
    <li><a href="index.html">PHP Manual</a></li>
    <li><a href="class.fiber.html">Fiber</a></li>
    <li>用一个异常来恢复 fiber 的执行</li>
  </ul>
</div>
<div id="layout">
  <div id="layout-content"><div id="fiber.throw" class="refentry">
 <div class="refnamediv">
  <h1 class="refname">Fiber::throw</h1>
  <p class="verinfo">(PHP 8 &gt;= 8.1.0)</p><p class="refpurpose"><span class="refname">Fiber::throw</span> &mdash; <span class="dc-title">用一个异常来恢复 fiber 的执行</span></p>

 </div>

 <div class="refsect1 description" id="refsect1-fiber.throw-description">
  <h3 class="title">说明</h3>
  <div class="methodsynopsis dc-description"><span class="modifier">public</span> <span class="methodname"><strong>Fiber::throw</strong></span>(<span class="methodparam"><span class="type"><a href="class.throwable.html" class="type Throwable">Throwable</a></span> <code class="parameter">$exception</code></span>): <span class="type"><a href="language.types.declarations.html#language.types.declarations.mixed" class="type mixed">mixed</a></span></div>

  <p class="para rdfs-comment">
   在当前调用 <span class="methodname"><a href="fiber.suspend.html" class="methodname">Fiber::suspend()</a></span> 的位置抛出指定的异常，恢复 fiber 执行。
  </p>
  <p class="para">
   如果调用此方法时 fiber 未挂起，将抛出 <span class="classname"><a href="class.fibererror.html" class="classname">FiberError</a></span>。
  </p>
 </div>


 <div class="refsect1 parameters" id="refsect1-fiber.throw-parameters">
  <h3 class="title">参数</h3>
  <dl>
   
    <dt>
<code class="parameter">exception</code></dt>

    <dd>

     <p class="para">
      在当前调用 <span class="methodname"><a href="fiber.suspend.html" class="methodname">Fiber::suspend()</a></span> 所在的位置，要抛出的异常。
     </p>
    </dd>

   
  </dl>

 </div>


 <div class="refsect1 returnvalues" id="refsect1-fiber.throw-returnvalues">
  <h3 class="title">返回值</h3>
  <p class="para">
   下次调用 <span class="methodname"><a href="fiber.suspend.html" class="methodname">Fiber::suspend()</a></span> 所提供的值；如果 fiber 已返回，则是 <strong><code>null</code></strong>。
   如果 fiber 暂停前抛出一个异常，它将会从调用本方法的位置抛出。
  </p>
 </div>


</div></div></div></body></html>